### ISSUE TITLE
Function fieldsForUpdate() bug with BelongsTo field

### ISSUE THREAD

**Issue by hnphi** [laravel/nova-issues#2580]
- Laravel Version: 7.9.2
- Nova Version: 3.5.0
- PHP Version: 7.4.3

### Description:
When I use `fieldsForUpdate()` to overwrite `fields()` in my **Resource**, I got error for **BelongsTo** field in **Edit** view.
![bug](https://user-images.githubusercontent.com/13552002/81161415-bcf13b80-8fb5-11ea-9d2f-6a8375dd83ea.png)

It's fine, if I use `BelongsTo::make('Teams', 'teams')` in both functions `fieldsForUpdate()` & `fields()`

### Steps To Reproduce:

- Create function `fieldsForUpdate()`
- Move **BelongsTo** field to `fieldsForUpdate()`
- Open **Edit** view, then error will show up

**Comment by omarmatijas** [comment]
Maybe I've found the origin of this "bug": in my case there was a problem with my .htacces. I have some sentences to make URLs lowercase. In my case, I have some relations with a single word (language) and other in camelcase (pageType). One relation worked right and the other failed... I've found that Nova got the request with "pagetype" and my relation was defined in camel case.
(I hope I made mysel clear; native spanish speaker).
